crates/factor-key-value/src/lib.rs

```diff
@@ -52,23 +49,10 @@ impl<R: RuntimeConfigResolver + 'static> Factor for KeyValueFactor<R> {
         &self,
         mut ctx: ConfigureAppContext<T, Self>,
     ) -> anyhow::Result<Self::AppState> {
-        // Build store manager from runtime config
-        let mut store_managers: HashMap<String, Arc<dyn StoreManager>> = HashMap::new();
-        if let Some(runtime_config) = ctx.take_runtime_config() {
-            for (store_label, config) in runtime_config.store_configs {
-                if let std::collections::hash_map::Entry::Vacant(e) =
-                    store_managers.entry(store_label)
-                {
-                    // Only add manager for labels that are not already configured. Runtime config
-                    // takes top-down precedence.
```

Comment:
I see the test asserting this is passing but not where this logic is ported to. Can you point it out to me?

Reply:
This is just verifying that we only ever have one store manager per label right? I don't see a test that asserts this. Can you point that out to me?

Reply:
Not exactly. You should be able to list multiple managers for the same label but the top one will be used (the variables provider works like this too). The `multiple_custom_key_value_uses_first_store` test checks this.

Reply:
Ah I think this is just a consequence of deserializing the toml. When toml is deserialized as a HashMap it just takes the first key (in this case the label `custom`) and ignores repeats. So none of the code we write explicitly does this, it happens automatically when we deserialize.